subiquitycore/file_util.py

```diff
@@ -73,3 +74,16 @@ def generate_config_yaml(filename, content, **kwargs):
         now = datetime.datetime.utcnow()
         tf.write(f'# Autogenerated by Subiquity: {now} UTC\n')
         tf.write(yaml.dump(content))
+
+
+def copy_file_if_exists(source, target):
+    """If source exists, copy to destination.  Ignore error that dest may be a
+    duplicate.  Create destination parent dirs as needed."""
+    if source is None or not os.path.exists(source):
```

Comment:
What would you think of marking `source` as a `str` and dropping the check for `source is None`?
I understand that it means making the call to `copy_file_if_exists` conditional (i.e., based on whether `self.autoinstall is None`) but I don't know about silently ignoring `None` values in this function.

Reply:
I'll do this if we don't end up with the symlink

Reply:
Alright, choice is up to you!

Reply:
Doing the non-symlink implementation, so I have done the above.